sdk/src/event_listener/substrate.rs

```diff
@@ -0,0 +1,58 @@
+use crate::event_listener::EventListener;
+use crate::Error;
+use async_trait::async_trait;
+use std::marker::PhantomData;
+use subxt::backend::StreamOfResults;
+use subxt::blocks::Block;
+use subxt::ext::sp_runtime::traits::Block as BlockT;
+use subxt::OnlineClient;
+use subxt_core::Config;
+use tokio::sync::Mutex;
+
+pub struct SubstrateEventListener<T: Config, Block: BlockT, Ctx> {
+    client: OnlineClient<T>,
+    context: Ctx,
+    stream: Mutex<StreamOfResults<subxt::blocks::Block<T, OnlineClient<T>>>>,
+    _pd: PhantomData<Block>,
+}
+
+#[async_trait]
+pub trait SubstrateEventListenerContext<T: Config>: Clone {
+    fn rpc_url(&self) -> String {
+        "ws://127.0.0.1:9944".to_string()
+    }
+
+    async fn handle_event(
+        &self,
+        event: Block<T, OnlineClient<T>>,
```

Comment:
Is this a block or event? Please see https://docs.rs/subxt/latest/subxt/events/struct.Events.html instead of https://docs.rs/subxt/latest/subxt/blocks/struct.Block.html, unless you mean here a block, not an event.

Reply:
The `Block`, with respect to the `EventListener` trait, is a type that represents an event. The `Events` struct and `Block` structs belong to a different context outside the scope of this one.

Reply:
It is confusing, the type is a Block, but you treat it as an event. If we are going all in blocks, then call it block not an event. Also, in your example, you did not show how to get the event? how to apply filters? ofc I know how, but you have to show them in the docs.